main/src/ca/uwaterloo/flix/language/phase/Weeder2.scala

```diff
@@ -3271,13 +3354,17 @@ object Weeder2 {
 
   /**
     * Picks out the first sub-tree of a specific [[TreeKind]].
+    *
+    * Returns a [[Tree]] of kind [[TreeKind.ErrorTree]] with `tree` as its only child.
     */
-  private def pick(kind: TreeKind, tree: Tree, synctx: SyntacticContext = SyntacticContext.Unknown): Validation[Tree, CompilationMessage] = {
+  private def pick(kind: TreeKind, tree: Tree, synctx: SyntacticContext = SyntacticContext.Unknown)(implicit sctx: SharedContext): Tree = {
     tryPick(kind, tree) match {
-      case Some(t) => Validation.success(t)
+      case Some(t) => t
       case None =>
-        val error = NeedAtleastOne(NamedTokenSet.FromTreeKinds(Set(kind)), synctx, loc = tree.loc)
-        Validation.HardFailure(Chain(error))
+        val error = NeedAtLeastOne(NamedTokenSet.FromTreeKinds(Set(kind)), synctx, loc = tree.loc)
+        sctx.errors.add(error)
+        val treeKind = TreeKind.ErrorTree(error)
+        Tree(treeKind, Array(tree), tree.loc.asSynthetic)
```

Comment:
This is the primary change and the reason for all the other changes. The idea here is just to return an error node that wraps the original tree.